Subject: [PKT_SCHED]: dsmark should ignore ECN bits
Date:	Tue, 28 Dec 2004 05:26:00 +0000
X-BK-ChangeSetKey: tgraf@suug.ch|ChangeSet|20041228052600|03888
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2034.116.34, 2004/12/27 21:26:00-08:00, tgraf@suug.ch

	[PKT_SCHED]: dsmark should ignore ECN bits
	
	Taking ECN bits into account doesn't make sense. The two bits were
	still unused at the time the code was written so this brings back the
	old behaviour.
	
	Signed-off-by: Thomas Graf <tgraf@suug.ch>
	Signed-off-by: David S. Miller <davem@davemloft.net>



 sch_dsmark.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)


diff -Nru a/net/sched/sch_dsmark.c b/net/sched/sch_dsmark.c
--- a/net/sched/sch_dsmark.c	2004-12-29 13:13:11 -08:00
+++ b/net/sched/sch_dsmark.c	2004-12-29 13:13:11 -08:00
@@ -14,6 +14,7 @@
 #include <linux/rtnetlink.h>
 #include <net/pkt_sched.h>
 #include <net/dsfield.h>
+#include <net/inet_ecn.h>
 #include <asm/byteorder.h>
 
 
@@ -198,10 +199,12 @@
 		/* FIXME: Safe with non-linear skbs? --RR */
 		switch (skb->protocol) {
 			case __constant_htons(ETH_P_IP):
-				skb->tc_index = ipv4_get_dsfield(skb->nh.iph);
+				skb->tc_index = ipv4_get_dsfield(skb->nh.iph)
+					& ~INET_ECN_MASK;
 				break;
 			case __constant_htons(ETH_P_IPV6):
-				skb->tc_index = ipv6_get_dsfield(skb->nh.ipv6h);
+				skb->tc_index = ipv6_get_dsfield(skb->nh.ipv6h)
+					& ~INET_ECN_MASK;
 				break;
 			default:
 				skb->tc_index = 0;
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

